core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnInsertExec.scala

```diff
@@ -156,7 +157,8 @@ case class ColumnInsertExec(_child: SparkPlan, partitionColumns: Seq[String],
        |  int $defaultRowSize = 0;
        |  ${declarations.mkString("\n")}
        |  $defaultBatchSizeTerm = Math.max(
-       |    (${math.abs(columnBatchSize)} - 8) / $defaultRowSize, 16);
+       |    (${math.abs( if (columnBatchSize > 0) columnBatchSize else Property.ColumnBatchSize.
+            defaultValue.get)} - 8) / $defaultRowSize, 16);
```

Comment:
Can you please verify from where columnBatchSize is being set to -1?

Two possible places are in ColumnFormatSamplingRelation:
      val buffer = batchCreator.createColumnBatchBuffer(
        // externalTableMetaData.columnBatchSize,
        // Fix for AQP-273, for sample table, the strata needs to remain as a single entry
        -1,

If so, good to change that code.